src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx

```diff
@@ -598,6 +601,51 @@ const getActivityConditionalLogic = (items: Item[]) =>
     return result;
   }, []);
 
+const getScoreConditions = (items: Item[], conditions?: Condition[]) =>
```

Comment:
if you provide `itemsObject` instead of `items` list you will reduce calculations in parent function `getScoresAndReports`